src/render.ts

```diff
@@ -98,6 +215,12 @@ export default function render(
  * @param config Optional config options
  * @param cb Callback function
  */
+export function renderAsync(
+    template: string | TemplateFunction,
+    data: object,
+    config?: PartialConfig,
+    cb?: CallbackFn
+): string | Promise<string> | void
```

Comment:
Think we should be able to delete lines 218-223, since those definitions are stated in lines 225-230.

Reply:
Oh, I think there might be a misunderstanding. Line 225-230 is the implementation signature, thus it will not be visible by our users.

> The signature of the implementation is not visible from the outside. [..]

See https://www.typescriptlang.org/docs/handbook/2/functions.html#function-overloads

If we want to make the implementation signature visible (for backwards compatibility in our case), we basically need to repeat it so that typescript knows we intend to make it visible to our users

Reply:
> The signature of the implementation is not visible from the outside.

@nhaef you're right, I didn't realize that Typescript had this behavior! You learn something new every day, I guess :)